### DISCUSSION TITLE
Keycloack authentication with Dart/Flutter

### DISCUSSION THREAD

**Discussion by elisavignali** [keycloak/keycloak#20841] [Q&A]
Hello,
I'm trying to authenticate my flutter app with keycloak (preferably  using keycloak_flutter library). I read a lot of documentation and topics on forums and did many trials, but I can't make it work. I tried with keycloak_flutter library and then without it, using  openid and flutter_appauth. I think I’m about to give up.
Has anyone done this and can give me some guidelines?

Elisa

**Comment by kirodh** [comment]
Hi 

Try one of these. We are currently also trying to figure this out.

https://medium.com/@rangika123.kanchana/keycloak-integration-for-flutter-web-using-openid-client-with-authorization-code-flow-489afeac6e9f

https://medium.com/@art.futurx/oauth-2-0-authentication-with-keycloak-in-flutter-applications-35d588b2246d

https://github.com/ronymesquita/flutter-and-keycloak/tree/main

http://code-addict.pl/flutter-spring-kotlin-web/
